workos/resources/user_management.py

```diff
@@ -1,232 +1,122 @@
-                response["impersonator"]
-            )
-            authentication_response.impersonator = impersonator
-        else:
-            authentication_response.impersonator = None
 
-        return authentication_response
+class Impersonator(WorkOSModel):
+    """Representation of a WorkOS Dashboard member impersonating a user"""
 
-    def to_dict(self):
-        authentication_response_dict = super(
-            WorkOSAuthenticationResponse, self
-        ).to_dict()
+    email: str
+    reason: str
 
-        user_dict = self.user.to_dict()
-        authentication_response_dict["user"] = user_dict
 
-        if self.impersonator:
-            authentication_response_dict["impersonator"] = self.impersonator.to_dict()
+class AuthenticationResponse(WorkOSModel):
+    """Representation of a User and Organization ID response as returned by WorkOS through User Management features."""
 
-        return authentication_response_dict
+    access_token: str
+    impersonator: Optional[Impersonator] = None
+    organization_id: Optional[str] = None
+    refresh_token: str
+    user: User
 
 
-class WorkOSRefreshTokenAuthenticationResponse(WorkOSBaseResource):
+class RefreshTokenAuthenticationResponse(WorkOSModel):
     """Representation of refresh token authentication response as returned by WorkOS through User Management features."""
 
-    """Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSRefreshTokenAuthenticationResponse comprises.
-    """
-
-    OBJECT_FIELDS = [
-        "access_token",
-        "refresh_token",
-    ]
-
-    @classmethod
-    def construct_from_response(cls, response):
-        authentication_response = super(
-            WorkOSRefreshTokenAuthenticationResponse, cls
-        ).construct_from_response(response)
-
-        return authentication_response
-
-    def to_dict(self):
-        authentication_response_dict = super(
-            WorkOSRefreshTokenAuthenticationResponse, self
-        ).to_dict()
-
-        return authentication_response_dict
-
-
-class WorkOSEmailVerification(WorkOSBaseResource):
-    """Representation of a EmailVerification object as returned by WorkOS through User Management features.
-
-    Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSEmailVerification comprises.
-    """
-
-    OBJECT_FIELDS = [
-        "id",
-        "user_id",
-        "email",
-        "expires_at",
-        "code",
-        "created_at",
-        "updated_at",
-    ]
-
-
-class WorkOSInvitation(WorkOSBaseResource):
-    """Representation of an Invitation as returned by WorkOS through User Management features.
-
-    Attributes:
-        OBJECT_FIELDS (list): List of fields a WorkOSInvitation comprises.
-    """
+    access_token: str
+    refresh_token: str
+
+
+class EmailVerification(WorkOSModel):
+    """Representation of a EmailVerification object as returned by WorkOS through User Management features."""
```

Comment:
Do we need to keep reminding developers that this is through User Management features?

Reply:
We don't. Removing that copy-pasta.